packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -16,51 +16,68 @@
 
 import '../style/index.css';
 
-import { ExpandableComponent } from '@elyra/ui-components';
 import {
-  ReactWidget,
-  UseSignal,
+  ExpandableComponent,
+  trashIcon,
+  importIcon
+} from '@elyra/ui-components';
+
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import {
   Clipboard,
   Dialog,
-  showDialog
+  showDialog,
+  ReactWidget,
+  UseSignal
 } from '@jupyterlab/apputils';
 import { CodeCell, MarkdownCell } from '@jupyterlab/cells';
-import { CodeEditor } from '@jupyterlab/codeeditor';
+import { CodeEditor, IEditorServices } from '@jupyterlab/codeeditor';
 import { PathExt } from '@jupyterlab/coreutils';
 import { IDocumentManager } from '@jupyterlab/docmanager';
 import { DocumentWidget } from '@jupyterlab/docregistry';
 import { FileEditor } from '@jupyterlab/fileeditor';
 import { Notebook, NotebookPanel } from '@jupyterlab/notebook';
 import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
-import { copyIcon, addIcon } from '@jupyterlab/ui-components';
+import { addIcon, copyIcon, editIcon } from '@jupyterlab/ui-components';
+
 import { Message } from '@lumino/messaging';
 import { Signal } from '@lumino/signaling';
 import { Widget } from '@lumino/widgets';
 
 import React from 'react';
 
 import { CodeSnippetService, ICodeSnippet } from './CodeSnippetService';
+import * as properties from './properties.json';
 
 /**
  * The CSS class added to code snippet widget.
  */
 const CODE_SNIPPETS_CLASS = 'elyra-CodeSnippets';
 const CODE_SNIPPETS_HEADER_CLASS = 'elyra-codeSnippetsHeader';
+const CODE_SNIPPETS_HEADER_BUTTON_CLASS = 'elyra-codeSnippetHeader-button';
 const CODE_SNIPPET_ITEM = 'elyra-codeSnippet-item';
 
+const METADATA_EDITOR_ID = 'elyra-metadata-editor';
+
+const defaultLanguages = ['Python', 'Java', 'R', 'Scala', 'Markdown'];
```

Comment:
How does this differ from the [`default_choices`](https://github.com/elyra-ai/elyra/blob/377f42df88a8e52201fa014961ff0e82b0a079bd/elyra/metadata/schemas/code-snippet.json#L33) in the schema file?  Seems like the schema should be used or the hint removed from the schema.

Reply:
I haven't finished that yet, but yes that will be how this works